### PR TITLE
Fix failures of OS Zoo CI

### PR DESCRIPTION
**Pull request by t8m**
Fix OS Zoo CI failures including replacing obsolete runner images.

### CODE REVIEW

File: .github/workflows/os-zoo.yml
Comment (paulidale) [inline .github/workflows/os-zoo.yml:36]: Do we really want all submodules?

Comment (t8m) [reply .github/workflows/os-zoo.yml:36]: This avoids error I've seen in previous runs when doing the submodule checkout explicitly - this is container so not so trivial to fix. Or I can just drop it and the fuzz/corpora won't be tested. IMO does not matter much.

**Comment by t8m** [comment]
Hmm the Windows CI failure is unrelated but still interesting. @hlandau https://github.com/openssl/openssl/actions/runs/5187386101/jobs/9349694642?pr=21131

File: .github/workflows/os-zoo.yml
Comment (paulidale) [reply .github/workflows/os-zoo.yml:36]: I'm leaning towards dropping it.  The fuzzing runs elsewhere, it's not going to provide anything extra here IMO.

Comment (t8m) [reply .github/workflows/os-zoo.yml:36]: OK, np.

**Review by paulidale — APPROVED** [review]

**Review by arapov — APPROVED** [review]
lgtm

**Comment by t8m** [comment]
ping @openssl/committers for second review

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to master branch. Thank you for the reviews.

**Pull request closed without merging**
